src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts

```diff
@@ -90,6 +90,14 @@ export abstract class AbstractFileDialogService implements IFileDialogService {
 			return this.preferredHome(schemeFilter);
 		}
 
+		const startParent = this.configurationService.getValue<boolean>('files.dialog.useParentPathForOpenFolder') !== false;
+		if (!startParent) {
+			try {
+				return (await this.fileService.stat(candidate)).isDirectory ? candidate : resources.dirname(candidate);
```

Comment:
Maybe easier to read if we check for `isDirectory` to return `candidate` and otherwise fall back to parent below?